src/lib.rs

```diff
@@ -80,6 +82,34 @@ impl<T> TemporalUnwrap for Option<T> {
     }
 }
 
+#[repr(i8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Sign {
+    Positive = 1,
+    Zero = 0,
+    Negative = -1,
+}
+
+impl From<i8> for Sign {
+    fn from(value: i8) -> Self {
+        match value.cmp(&0) {
+            Ordering::Greater => Self::Positive,
+            Ordering::Equal => Self::Zero,
+            Ordering::Less => Self::Negative,
+        }
+    }
+}
+
+impl Sign {
+    /// Coerces the current `Sign` to be either negative or positive.
+    pub(crate) fn as_non_zero(&self) -> Self {
+        if matches!(self, Self::Zero) {
+            return Self::Positive;
+        }
+        *self
+    }
```

Comment:
If this is to multiply a duration with the sign, what about having an `as_sign_multiplier` function instead that returns `1i8` for zero and positive, and `-1i8` for negative?

Reply:
It is. It's specifically a nonzero variant in these rounding methods vs. the regular multiplication that occurs elsewhere. I'm fine with `as_sign_multiplier` though. Better than anything I'm coming up with in this instance.